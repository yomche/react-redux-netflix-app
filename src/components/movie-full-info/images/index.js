export { default as headerBackground } from './netflix-background.jpg';