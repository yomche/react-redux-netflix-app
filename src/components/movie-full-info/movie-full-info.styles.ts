import styled from 'styled-components';

export const StyledMovieFullInfoHeader = styled.header`
  width: 1200px;
  height: 520px;
  margin: auto;
`;
