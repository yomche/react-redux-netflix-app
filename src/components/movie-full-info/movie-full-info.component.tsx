import React, { FC } from 'react';
import { StyledMovieFullInfoHeader } from './movie-full-info.styles';
import { MovieCardFullSize } from '../movie-card-full-size/movie-card-full-size.component';
import { CurrentMovieType } from '../../types';

type MovieFullInfoProps = {
  currentMovieFullInfo: CurrentMovieType;
};

export const MovieFullInfo: FC<MovieFullInfoProps> = ({ currentMovieFullInfo }) => (
  <StyledMovieFullInfoHeader>
    <MovieCardFullSize
      poster_path={currentMovieFullInfo.poster_path}
      title={currentMovieFullInfo.title}
      release_date={currentMovieFullInfo.release_date}
      overview={currentMovieFullInfo.overview}
      key={currentMovieFullInfo.id}
    />
  </StyledMovieFullInfoHeader>
);
