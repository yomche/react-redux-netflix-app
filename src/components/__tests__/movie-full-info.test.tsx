import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import { MovieFullInfo } from '../movie-full-info/movie-full-info.component';

describe('< MovieFullInfo /> component', () => {
  const filmMock = {
    poster_path: 'Test URL',
    title: 'Test title',
    release_date: 'Test release date',
    overview: 'Test overview',
    id: 0,
  };

  test('renders correctly', () => {
    const { asFragment } = render(
      <Router>
        <MovieFullInfo currentMovieFullInfo={filmMock} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
