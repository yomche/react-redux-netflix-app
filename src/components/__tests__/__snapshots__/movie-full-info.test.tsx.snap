// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< MovieFullInfo /> component renders correctly 1`] = `
<DocumentFragment>
  <header
    class="StyledMovieFullInfoHeader-hsjhx9-0 ByEMN"
  >
    <section
      class="StyleFullMovieCard-sc-1y5zepo-0 eGohNm"
    >
      <img
        class="StyledMovieCardPoster-sc-1y5zepo-1 cLMjai"
        src="Test URL"
      />
      <div
        class="StyledMovieCardInfo-sc-1y5zepo-2"
      >
        <h1
          class="StyledHeading-sc-1y5zepo-3 kNlvlE"
        >
           Test title 
        </h1>
        <p
          class="StyledText-sc-1y5zepo-4 clrOEw"
        >
           Test release date 
        </p>
        <p
          class="StyledText-sc-1y5zepo-4 clrOEw"
        >
           Test overview 
        </p>
      </div>
    </section>
  </header>
</DocumentFragment>
`;
